validator/validation/graphics.py

```diff
@@ -246,42 +253,43 @@ def generate_all_graphs(validation_run, outfolder):
     # get units for plot labels
     unit_ref = validation_run.reference_configuration.dataset.short_name
 
+    ds =  netCDF4.Dataset(validation_run.output_file.path, mode='r')
```

Comment:
Is there a particular reason why you removed the 'with' keyword and close manually at the end? My gut feeling is that leaving a netcdf file hanging without closing would be a prime candidate for causing a segfault later on. Since you close at the end, it shouldn't make a difference - unless you leave the block somewhere before the end, e.g. when an error occurs.
Could you test if the segfault goes away when you open the netcdf dataset using 'with'?

Reply:
thats just the way how I usually do it without getting segfaults ;-) The file gets closed also upon a crash, so it should never stay opened... I don't think it makes a difference, therefore I have no problem reverting this if the old way turns out to work as well.

Reply:
Yeah, I also suspect it doesn't or shouldn't make a difference, I just learned that using 'with' was more pythonic than not doing it.